### ISSUE TITLE
Multi scale training reduces accuracy

### ISSUE THREAD

**Issue by CodeJjang** [jwyang/faster-rcnn.pytorch#103]
After successfuly training on my custom 600x900 dataset (with the tips you said, bs = 1, set 'need_crop' to False), I have added some 4kX2k images to my dataset which contains more objects.

But instead of raising my accuracy, it decreased it:
AP for large vehicle = 0.3520 (was 75)
AP for small vehicle = 0.7978 (was 89)
AP for solar panel = 0.0529
Mean AP = 0.4009

I thought it was because I set 'need_crop' to False, so what happened was that 4k images was rescaled to 600 (default scale size), but the roi_db wasn't cropped, so it messed the GT bounding boxes.
Am I right?

So, I removed the line that sets 'need_crop' to False, but then I receieved errors again (the NaN errors, because fg/bg was 128/0):
[session 1][epoch  1][iter  500] loss: 0.8761, lr: 4.00e-03
			fg/bg=(32/96), time cost: 324.625776
			rpn_cls: 0.0196, rpn_box: 0.0169, rcnn_cls: 0.3088, rcnn_box 0.3363
[session 1][epoch  1][iter  600] loss: nan, lr: 4.00e-03
			fg/bg=(128/0), time cost: 324.277149
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan

So I guess I need to set again 'need_crop' to False, but how do I deal with the fact that the 4k images ruined my accuracy instead of helping it? Tried also to train with SCALE=600,1600 (max my GPU could handle), but the accuracies are still worse than the original 600x900 images experiment.
And also, can you explain why bs=1 was related?

**Comment by CodeJjang** [comment]
Thanks.  

> So you can directly rescale the image to a single size and then train the object detector.
You mean `crop`, not `rescale`, right? So I'll try to crop the high res images to something like 600x600s and then feed them again. Thanks

**Comment by jwyang** [comment]
> So if I use batch size == 1, it's useless right?

Yes, that's right.

> But can you explain me why you suggested using batch size 1?

that's fine to set batch size > 1

> You mean `crop`, not `rescale`, right? So I'll try to crop the high res images to something like 600x600s and then feed them again.

Yes, in this case, you will have relatively bigger objects. if the relative scale of object in your 4kx2k images is similar to your smaller size images, then you do not need to crop them, just rescale them. That's should be fine. Also check whether your added higher resolution images are in the same domain of your smaller size images. This will also affect the performance actually.

**Comment by jwyang** [comment]
@CodeJjang It seems that you have addressed this issue. I am closing it for now.
